test/rules/implicit_reopen_test.dart

```diff
@@ -0,0 +1,195 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:test_reflective_loader/test_reflective_loader.dart';
+
+import '../rule_test_support.dart';
+
+main() {
+  defineReflectiveSuite(() {
+    defineReflectiveTests(ImplicitReopenTest);
```

Comment:
Just a drive-by comment to also add tests for mixin applications:

```dart
final mixin M {}
class C = Object with M {}
```

Reply:
Can't help correcting it a little heh.
```dart
class C = Object with M;
```

Reply:
Ah!  Class type aliases...  Good catch!